services-custom/dynamodb-enhanced/src/test/java/software/amazon/awssdk/enhanced/dynamodb/ProjectionExpressionConvertorTest.java

```diff
@@ -0,0 +1,74 @@
+package software.amazon.awssdk.enhanced.dynamodb;
+
+import org.junit.Assert;
+import org.junit.Test;
+import software.amazon.awssdk.enhanced.dynamodb.internal.ProjectionExpressionConvertor;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+
+public class ProjectionExpressionConvertorTest {
+
+    public static final String MAPPED_INDICATOR = "#AMZN_MAPPED_";
+    public static final String NESTING_SEPARATOR = ".";
+
+    @Test
+    public void testAttributeNameWithNoNestedAttributes() {
+        final String keyName = "fieldKey";
+        NestedAttributeName attributeName = NestedAttributeName.builder().elements(keyName).build();
+        ProjectionExpressionConvertor expressionConvertor = ProjectionExpressionConvertor.create(Arrays.asList(attributeName));
+        final Map<String, String> stringStringMap = expressionConvertor.convertToExpressionMap();
+        final Optional<String> toNameExpression = expressionConvertor.convertToProjectionExpression();
+        Map<String, String> expectedmap = new HashMap<>();
+        expectedmap.put(MAPPED_INDICATOR + keyName, keyName);
+        Assert.assertEquals(expectedmap, stringStringMap);
+        Assert.assertEquals(MAPPED_INDICATOR + keyName, toNameExpression.get());
```

Comment:
Our assertions libraries in order of preference are : assertJ, hamcrest, junit. You picked the least preferential, can we use assertJ here and everywhere else you have created new test files.

Reply:
updated